### PR TITLE
#181339469 setting-up-unit-tests

### PR DESCRIPTION
**Pull request by blandri**
### What does this PR do?
Set up unit tests for the project

### Description of Task to be completed?

- add packages that support testing
- add tests of welcome routes as reference

### How should this be manually tested?

- Clone this repository and open the project in your code editor.
- Check out  [chore/#181339469-set-up-unit-test](https://github.com/atlp-rwanda/cabal-bn-be/tree/chore/%23181339469-set-up-unit-test)
- to install all package on your local repo use: **yarn install** 
- to run the project use: **yarn run test**

### What are the relevant pivotal tracker stories?
[#181339469](https://www.pivotaltracker.com/n/projects/2555280)
### Screenshots (if appropriate)
image

### CODE REVIEW

File: .nyc_output/e0761902-7651-4daf-8e15-23c2f36343e5.json [hunk 1]
```diff
@@ -0,0 +1 @@
...{"line":10,"column":94}}],"line":10},"1":{"loc":{"start":{"line":10,"column":46},"end":{"line":10,"column":67}},"type":"binary-expr","locations":[{"start":{"line":10,"column":46},"end":{"line":10,"column":49}},{"start":{"line":10,"column":53},"end":{"line":10,"column":67}}],"line":10}},"s":{"0":1,"1":1,"2":1,"3":1,"4":1,"5":1,"6":0,"7":1,"8":1},"f":{"0":1,"1":0},"b":{"0":[0,1],"1":[1,1]},"inputSourceMap":{"version":3,"sources":["homeRoute.js"],"names":["home","express","Router","get","req","res","next","send"],"mappings":";;;;;;;AAAA;;;;AAEA,IAAMA,IAAI,GAAGC,oBAAQC,MAAR,EAAb;;AAEAF,IAAI,CAACG,GAAL,CAAS,GAAT,EAAc,UAACC,GAAD,EAAMC,GAAN,EAAWC,IAAX,EAAoB;AAChCD,EAAAA,GAAG,CAACE,IAAJ,CAAS,yBAAT;AACD,CAFD;eAIeP,I","sourceRoot":"C:\\Users\\landr\\Desktop\\cabal-bn-be\\src\\routes\\api\\","sourcesContent":["import express from 'express';\r\n\r\nconst home = express.Router();\r\n\r\nhome.get('/', (req, res, next) => {\r\n  res.send('Welcome to our homepage');\r\n});\r\n\r\nexport default home;\r\n"]},"_coverageSchema":"1a1c01bbd47fc00a2c39e90264f33305004495a9","hash":"c1dd48242968f340093e6ed93a78acdb4334412e","contentHash":"a2b1d82e262d719660df948f8e8da0ad09c53367614dd5eb0c43ed6b9c498187"}}
```
Comment (Tsa2341) [inline .nyc_output/e0761902-7651-4daf-8e15-23c2f36343e5.json]: don't push .nyc files

File: coverage/app.js.html [hunk 2]
```diff
@@ -0,0 +1,124 @@
+
```
Comment (Tsa2341) [inline coverage/app.js.html]: I also, think coverage related files are not to be pushed

File: .nyc_output/e0761902-7651-4daf-8e15-23c2f36343e5.json (hunk 1 above)
Comment (Code-250) [inline .nyc_output/e0761902-7651-4daf-8e15-23c2f36343e5.json]: please do not push nyc_output folder

File: coverage/app.js.html [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            <div class='footer quiet pad2 space-top1 center small'>
+                Code coverage generated by
+                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
+                at Mon Feb 28 2022 16:40:10 GMT+0200 (Central Africa Time)
+            </div>
+        <script src="prettify.js"></script>
+        <script>
+            window.onload = function () {
+                prettyPrint();
+            };
```
Comment (Code-250) [inline coverage/app.js.html]: do not push coverage related folder

**Review by Code-250 — CHANGES_REQUESTED** [review]
work on all requested changes.

**Review by Gabin-ishimwe — CHANGES_REQUESTED** [review]
coverage folders are not needed in pushing

**Review by niyonkurueric — APPROVED** [review]

**Review by Tsa2341 — CHANGES_REQUESTED** [review]
please work on all comments.

if you go to your branch on Github you will see that the coverage and .nyc files are still in.
adding them in .gitignore will not delete them as it only prevents but does not delete already pushed code.

use [ this article ](https://dev.to/morinoko/useful-git-commands-for-removing-accidentally-pushed-or-committed-files-2ld) for reference on how to delete already pushed folder or file.

**Review by Tsa2341 — APPROVED** [review]

**Review by Code-250 — APPROVED** [review]

**Review by Gabin-ishimwe — APPROVED** [review]

**Review by ncutixavier — COMMENTED** [review]
Add Circle CI & Coveralls badges

**Comment by ncutixavier** [comment]
@blandri  describe well how to test your code locally

**Pull request merged**